Dockerfile

```diff
@@ -0,0 +1,25 @@
+# syntax=docker/dockerfile:1
+FROM --platform=$BUILDPLATFORM golang:1.17 as build
+ARG TARGETPLATFORM
+ARG BUILDPLATFORM
+
```

Comment:
I would add an `ARG DEBIAN_FRONTEND=noninteractive` statement here to make it available to later `apt` commands.

Reply:
ENV shoud be better

Reply:
I used to do `ENV` until I discovered that using `ARG` does not persist the variable setting after the build. If you `docker exec` into the container (ie. _interactively_). If `ENV` is used, you'll have it set while doing `exec` interactively, so you have to unset it. If `ARG`, you won't have it set when interactive.